core/src/mindustry/logic/LAssembler.java

```diff
@@ -61,8 +61,14 @@ public static String write(Seq<LStatement> statements){
         String[] lines = data.split("\n");
         int index = 0;
         for(String line : lines){
-            //comments
-            if(line.startsWith("#") || line.isEmpty()) continue;
+            // comments
+            int commentIdx = line.indexOf('#');
+            if (commentIdx != -1){
+                line = line.substring(0, commentIdx).trim();
+            }
+            if (line.isEmpty()){
+                continue;
+            }
```

Comment:
Make this a one-liner like in the code you replaced.
```suggestion
            if(line.isEmpty()) continue;
```
Also, in line 66, remove the space in `if (` to match Mindustry code style.